### ISSUE TITLE
Getting user's email with LinkedIn

### ISSUE THREAD

**Issue by scozma** [sarfarazansari/ng4-social-login#5]
Hi,

Can you please advise if it's possible to retrieve also the user's email address while logging in with LinkedIn?
I see the email is available while logging in with Facebook and Google. Is this possible also with LinkedIn, or they don't make the email address public for their users?

Thanks!

**Comment by sarfarazansari** [comment]
As per my Knowledge, they don't provide Email Id. Am also looking for that. But what I found on Docs that they don't provide.
for more info, you can check [https://developer.linkedin.com/docs/getting-started-js-sdk](https://developer.linkedin.com/docs/getting-started-js-sdk)

**Comment by scozma** [comment]
@sarfarazansari - Thanks for your response. Sorry, but I don't understand your message - did you fix this, or should I do something regarding it? I installed the package again, but the emails is still 'undefined' for LinkedIn. Please advise. Thanks!

**Comment by sarfarazansari** [comment]
@scozma The feature is already implemented in Plugin.
make sure when you generating app key from linkedin. APP Key have all permissions.
you'll see an option when you generate an app from linkedin developer platform.

I don't need to do extra thing.

![image](https://user-images.githubusercontent.com/3416989/33071551-011dd7a6-cee2-11e7-8678-67b7c2c72551.png)

open developer console and look for `https://api.linkedin.com/v1/people/~:(id,first-name,last-name,email-address,picture-url)` in Response of this API you'll definitely get LinkedInResponse.

I think the main problem you are facing will be Linkedin App key generation. Make sure you follow all steps.

If still you are not able to get. then please create a plunkr or js fiddle

**Comment by scozma** [comment]
@sarfarazansari - Indeed, my login screen didn't show the "YOUR EMAIL ADDRESS" part, it just showed "YOUR PROFILE OVERVIEW".
I went to https://developer.linkedin.com/ and accessed my app. On the "Authentification" tab, I had to check also the **r_emailaddress** checkbox from below "Default Application Permissions". Now everything works fine and I can receive the user's email address.

Many thanks for your help!
